### PR TITLE
Merge new test-infrastructure into develop

### PR DESCRIPTION
**Pull request by cgmillette**

### CODE REVIEW

File: test/01_contraction/CMakeLists.txt [hunk 1]
```diff
@@ -23,14 +23,32 @@
...
+add_hiptensor_test(bilinear_contraction_f64_test ${BilinearContractionF64TestSources})
+add_hiptensor_test(scale_contraction_f32_test ${ScaleContractionF32TestSources})
+add_hiptensor_test(scale_contraction_f64_test ${ScaleContractionF64TestSources})
+
 # Brute force tests
-add_hiptensor_test(scale_contraction_f32_test scale_contraction_f32.cpp)
-add_hiptensor_test(scale_contraction_f64_test scale_contraction_f64.cpp)
-add_hiptensor_test(bilinear_contraction_f32_test bilinear_contraction_f32.cpp)
-add_hiptensor_test(bilinear_contraction_f64_test bilinear_contraction_f64.cpp)
+# add_hiptensor_test(scale_contraction_f32_test scale_contraction_f32.cpp)
```
Comment (dlangbe) [inline test/01_contraction/CMakeLists.txt]: Can remove old tests

File: test/01_contraction/bilinear_contraction_ac_f32.cpp
Comment (dlangbe) [inline test/01_contraction/bilinear_contraction_ac_f32.cpp:1]: We can remove this and all other old test files entirely

File: test/01_contraction/common.hpp [hunk 2]
```diff
@@ -34,13 +34,14 @@
...
 #include <unordered_map>
+#include <vector>
 
 // hiptensor includes
 #include <hiptensor/hiptensor.hpp>
 #include <hiptensor/hiptensor_types.hpp>
 #include <hiptensor/internal/hiptensor_utility.hpp>
 
-#include "contraction_cpu_reference.hpp"
+// #include "contraction_cpu_reference.hpp"
```
Comment (dlangbe) [inline test/01_contraction/common.hpp]: Can remove

**Review by dlangbe — DISMISSED** [review]
LGTM!

**Review by dlangbe — APPROVED** [review]

File: test/hiptensor-test-yaml.hpp.in [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ *******************************************************************************/
+
+#ifndef HIPTENSOR_TEST_ @hiptensor_test_NAME @_YAML_HPP
```
Comment (CongMa13) [inline test/hiptensor-test-yaml.hpp.in]: - There are whitespaces after HIPTENSOR_TEST_ and hiptensor_test_NAME. Is it a typo?

Comment (cgmillette) [reply test/hiptensor-test-yaml.hpp.in]: Yes - clang formatter got me. Fixed.

**Pull request merged**